packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx

```diff
@@ -119,14 +165,109 @@ describe('ResourceModal', () => {
+    history = createMemoryHistory({ getUserConfirmation });
+
+    const props = {
+      modalTitle: editModalInfo.title,
+      modalDescription: editModalInfo.description,
+      type: 'Note' as const,
+      title: 'A title',
+      description: 'A description',
+    };
+
+    render(
+      <Router history={history}>
+        <ResourceModal {...defaultProps} {...props} />
+      </Router>,
+    );
+    enterType('Link');
+    enterLink('http://example.com');
+    enterTitle('A new title');
+    enterDescription('A new description');
+    const cancelButton = screen.getByRole('link', { name: /cancel/i });
+    userEvent.click(cancelButton);
+    expect(getUserConfirmation).toHaveBeenCalledTimes(1);
+  });
+
+  it("the dialog doesn't show when the user doesn't add new changes", () => {
+    getUserConfirmation = jest.fn((_message, cb) => cb(true));
+    history = createMemoryHistory({ getUserConfirmation });
+
+    const props = {
+      modalTitle: editModalInfo.title,
+      modalDescription: editModalInfo.description,
+      type: 'Note' as const,
+      title: 'A title',
+      description: 'A description',
+    };
+
+    render(
+      <Router history={history}>
+        <ResourceModal {...defaultProps} {...props} />
+      </Router>,
+    );
+    const cancelButton = screen.getByRole('link', { name: /cancel/i });
+    userEvent.click(cancelButton);
+    expect(getUserConfirmation).not.toHaveBeenCalled();
+  });
+  it('the dialog shows when the user changes type of resource', () => {
+    getUserConfirmation = jest.fn((_message, cb) => cb(true));
+    history = createMemoryHistory({ getUserConfirmation });
+
+    const props = {
+      modalTitle: editModalInfo.title,
+      modalDescription: editModalInfo.description,
+      type: 'Note' as const,
+      title: 'A title',
+      description: 'A description',
+    };
+
+    render(
+      <Router history={history}>
+        <ResourceModal {...defaultProps} {...props} />
+      </Router>,
+    );
+    enterType('Link');
+    const cancelButton = screen.getByRole('link', { name: /cancel/i });
+    userEvent.click(cancelButton);
+    expect(getUserConfirmation).toHaveBeenCalledTimes(1);
+  });
+  it('the dialog shows when the user changes the link', () => {
+    getUserConfirmation = jest.fn((_message, cb) => cb(true));
+    history = createMemoryHistory({ getUserConfirmation });
+
+    const props = {
+      modalTitle: editModalInfo.title,
+      modalDescription: editModalInfo.description,
+      type: 'Link' as const,
+      externalLink: 'http://example.com',
+      title: 'A title',
+      description: 'A description',
+    };
+
+    render(
+      <Router history={history}>
+        <ResourceModal {...defaultProps} {...props} />
+      </Router>,
+    );
+    enterLink('http://example2.com');
+    const cancelButton = screen.getByRole('link', { name: /cancel/i });
+    userEvent.click(cancelButton);
+    expect(getUserConfirmation).toHaveBeenCalledTimes(1);
+  });
```

Comment:
Missing tests to see if we correctly render a resource.

i.e. is the title displayed correctly etc.